packages/api/example/ExampleStorage.ts

```diff
@@ -0,0 +1,71 @@
+import { AsyncStorage, StorageKeys, StorageModel } from "../src";
+import { ExampleExtension, MyExampleExtension } from "./ExampleExtension";
+
+type MyModel = StorageModel<{
+  age: number;
+  name: string;
+  likes: boolean[];
+}>;
+
+// @ts-ignore
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+class ExampleStorage implements AsyncStorage<MyModel, MyExampleExtension> {
+  private storage: MyModel = {
+    age: null,
+    name: null,
+    likes: null,
+  };
+
+  getItem = async <K extends keyof MyModel>(key: K): Promise<MyModel[K]> => {
+    return this.storage[key];
+  };
+
+  setItem = async <K extends StorageKeys<MyModel>>(
+    key: K,
+    value: MyModel[K]
+  ): Promise<void> => {
+    this.storage[key] = value;
+  };
+
+  removeItem = async <K extends StorageKeys<MyModel>>(
+    key: K
+  ): Promise<void> => {
+    this.storage[key] = null;
+  };
+
+  getMany = async <K extends StorageKeys<MyModel>>(
+    keys: K[]
+  ): Promise<{ [k in K]: MyModel[k] }> => {
+    return keys.reduce((entries, key) => {
+      return {
+        ...entries,
+        [key]: this.storage[key] ?? null,
+      };
```

Comment:
Nit: We should mutate instead of copying in a loop.

Reply:
Instead of using the spread op every iteration:
```suggestion
    return keys.reduce((entries, key) => {
      entries[key] = this.storage[key] ?? null;
      return entries;
```